applications/FluidDynamicsApplication/python_scripts/adjoint_vmsmonolithic_solver.py

```diff
@@ -100,17 +100,30 @@ def Initialize(self):
         self.GetResponseFunction().Initialize()
         self.GetSensitivityBuilder().Initialize()
 
+        domain_size = self.main_model_part.ProcessInfo[KratosMultiphysics.DOMAIN_SIZE]
+        KratosMultiphysics.NormalCalculationUtils().CalculateOnSimplex(self.main_model_part.Conditions, domain_size)
+        KratosMultiphysics.NormalCalculationUtils().CalculateNormalShapeDerivativesOnSimplex(self.main_model_part.Conditions, domain_size)
+
         KratosMultiphysics.Logger.PrintInfo(self.__class__.__name__, "Solver initialization finished.")
 
     def _CreateScheme(self):
         response_function = self.GetResponseFunction()
         scheme_type = self.settings["scheme_settings"]["scheme_type"].GetString()
+        domain_size = self.main_model_part.ProcessInfo[KratosMultiphysics.DOMAIN_SIZE]
         if scheme_type == "bossak":
-            scheme = KratosMultiphysics.ResidualBasedAdjointBossakScheme(
-                self.settings["scheme_settings"],
-                response_function)
+            if domain_size == 2:
+                scheme = KratosCFD.VelocityBossakAdjointScheme2D(self.settings["scheme_settings"], response_function)
+            elif domain_size == 3:
+                scheme = KratosCFD.VelocityBossakAdjointScheme3D(self.settings["scheme_settings"], response_function)
```

Comment:
Again I refer to the domain size templating... this differs with the general Kratos schemes line.

Reply:
done :)